src/vs/workbench/api/node/extHostLogService.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { TPromise } from 'vs/base/common/winjs.base';
+import { mkdirp, dirExists } from 'vs/base/node/pfs';
+import Event, { Emitter } from 'vs/base/common/event';
+import { ExtHostLogServiceShape } from './extHost.protocol';
+import { LogLevel } from 'vs/workbench/api/node/extHostTypes';
+import { ILogService } from 'vs/platform/log/common/log';
+import { createLogService } from 'vs/platform/log/node/spdlogService';
+import { IEnvironmentService } from 'vs/platform/environment/common/environment';
+
+export class ExtHostLogService implements ExtHostLogServiceShape {
+	private _loggers: Map<string, ExtHostLogger> = new Map();
+
+	private _onDidChangeLogLevel: Emitter<LogLevel>;
+	get onDidChangeLogLevel(): Event<LogLevel> { return this._onDidChangeLogLevel.event; }
+
+	constructor(private _environmentService: IEnvironmentService) {
+		this._onDidChangeLogLevel = new Emitter<LogLevel>();
+	}
+
+	$acceptLogLevelChanged(logLevel: LogLevel): void {
+		this._onDidChangeLogLevel.fire(logLevel);
+	}
+
+	getExtLogger(extensionID: string): ExtHostLogger {
+		if (!this._loggers.has(extensionID)) {
+			const logService = createLogService(extensionID, this._environmentService, extensionID);
+			const logsDirPath = path.join(this._environmentService.logsPath, extensionID);
+			this._loggers.set(extensionID, new ExtHostLogger(this, logService, logsDirPath));
+		}
+
+		return this._loggers.get(extensionID);
+	}
+}
+
+export class ExtHostLogger implements vscode.ILogger {
+	private _currentLevel: LogLevel;
+
+	get onDidChangeLogLevel(): Event<LogLevel> { return this._extHostLogService.onDidChangeLogLevel; }
+
+	constructor(
+		private readonly _extHostLogService: ExtHostLogService,
+		private readonly _logService: ILogService,
+		private readonly _logDirectory: string
```

Comment:
Not really sure I understand but it seems like one logger-instance for all extensions but a different log directory for each. Is that correct and iff so is that desired. Should each extension get its own log file? Should each log message be prefixed with the extension ID otherwise?

Reply:
I think extension authors would be annoyed at having to read logs with messages from other extensions mixed in. Some extensions will be spammy.

Reply:
Oh and to explain more, one ExtHostLogger per extension. Each has its own ILogService. ExtHostLogService creates the loggers.